java.lang.UnsupportedOperationException: Unsupported MySQL version: 11.4.7-4-MariaDB-enterprise-log

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

java.lang.UnsupportedOperationException: Unsupported MySQL version: 11.4.7-4-MariaDB-enterprise-log

### SeaTunnel Version

/root/apache-seatunnel-2.3.11/config/

### SeaTunnel Config

```conf
env {
  parallelism = "1"
  job.mode = "BATCH"
}

source {
  Jdbc {
    url = "jdbc:mysql://10.2.19.214:53306/ics_front"
    driver = "com.mysql.cj.jdbc.Driver"
    connection_check_timeout_sec = 3000
    user = "x2doris"
    password = "***************"
    query = """
      SELECT * FROM igtjournal where tran_date = '20251021'
    """
  }
}


sink {
    Doris {
        fenodes = "10.1.20.62:8030"
        username = "core_import"
        password = "********"
        database = "topatdb"
        table = "igtjournal"
        table.identifier = "topatdb.igtjournal"
        sink.enable-2pc = "true"
        sink.label-prefix = "igtjournal"
        doris.config = {
          format = "json"
          read_json_by_line = "true"
        }
    }
}
```

### Running Command

```shell
./bin/seatunnel.sh --config ./config/igtjournal.conf
```

### Error Exception

```log
Caused by: com.mysql.cj.exceptions.CJCommunicationsException: Communications link failure
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

java.lang.UnsupportedOperationException: Unsupported MySQL version: 11.4.7-4-MariaDB-enterprise-log

### Screenshots

<img width="1123" height="441" alt="Image" src="https://github.com/user-attachments/assets/12b3047b-76fd-41ea-add9-44a594f49ea1" />

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/docs/en/connectors/sink/Jdbc.md
+++ b/docs/en/connectors/sink/Jdbc.md
@@ -131,6 +131,19 @@ Expected rows:
 
 If the job fails before writing, check [Troubleshooting](#troubleshooting) first.
 
+:::note
+
+When connecting to MariaDB, use MariaDB Connector/J with the matching URL and driver:
+
+```hocon
+url = "jdbc:mariadb://localhost:3306/database"
+driver = "org.mariadb.jdbc.Driver"
+```
+
+Do not use MySQL Connector/J with a `jdbc:mysql:` URL for MariaDB. That configuration selects the MySQL dialect, which can reject a MariaDB server version as an unsupported MySQL version.
+
+:::
+
 ## Key Features
 
 - [x] [exactly-once](../../introduction/concepts/connector-v2-features.md)

--- a/docs/en/connectors/source/Jdbc.md
+++ b/docs/en/connectors/source/Jdbc.md
@@ -120,6 +120,19 @@ cd "${SEATUNNEL_HOME}"
 
 If the job fails before reading rows, check [Troubleshooting](#troubleshooting) first.
 
+:::note
+
+When connecting to MariaDB, use MariaDB Connector/J with the matching URL and driver:
+
+```hocon
+url = "jdbc:mariadb://localhost:3306/database"
+driver = "org.mariadb.jdbc.Driver"
+```
+
+Do not use MySQL Connector/J with a `jdbc:mysql:` URL for MariaDB. That configuration selects the MySQL dialect, which can reject a MariaDB server version as an unsupported MySQL version.
+
+:::
+
 ## Key features
 
 - [x] [batch](../../introduction/concepts/connector-v2-features.md)

--- a/docs/zh/connectors/sink/Jdbc.md
+++ b/docs/zh/connectors/sink/Jdbc.md
@@ -131,6 +131,19 @@ ORDER BY id;
 
 如果任务在写入前失败，请先检查[故障排查](#故障排查)。
 
+:::note
+
+连接 MariaDB 时，请使用 MariaDB Connector/J 以及匹配的 URL 和驱动：
+
+```hocon
+url = "jdbc:mariadb://localhost:3306/database"
+driver = "org.mariadb.jdbc.Driver"
+```
+
+不要使用 MySQL Connector/J 和 `jdbc:mysql:` URL 连接 MariaDB。该配置会选择 MySQL 方言，可能将 MariaDB 服务端版本判定为不支持的 MySQL 版本。
+
+:::
+
 ## 主要特性
 
 - [x] [精确一次](../../introduction/concepts/connector-v2-features.md)

--- a/docs/zh/connectors/source/Jdbc.md
+++ b/docs/zh/connectors/source/Jdbc.md
@@ -120,6 +120,19 @@ cd "${SEATUNNEL_HOME}"
 
 如果任务在读取数据前失败，请先检查[故障排查](#故障排查)。
 
+:::note
+
+连接 MariaDB 时，请使用 MariaDB Connector/J 以及匹配的 URL 和驱动：
+
+```hocon
+url = "jdbc:mariadb://localhost:3306/database"
+driver = "org.mariadb.jdbc.Driver"
+```
+
+不要使用 MySQL Connector/J 和 `jdbc:mysql:` URL 连接 MariaDB。该配置会选择 MySQL 方言，可能将 MariaDB 服务端版本判定为不支持的 MySQL 版本。
+
+:::
+
 ## 关键特性
 
 - [x] [批](../../introduction/concepts/connector-v2-features.md)

